### PR TITLE
test: Add content validation to Agent and Agent Api NuGets.

### PR DESCRIPTION
**Pull request by nrcventura**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

Adds content validation to the Agent NuGet package and the Agent Api NuGet package. Not everything in the nuget packages were validated. Instead, the focus is on validating the components that are defined in the nuspec files.

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: build/ArtifactBuilder/Artifacts/NugetAgent.cs [hunk 1]
```diff
@@ -95,5 +99,99 @@ private void TransformNewRelicConfig(string newRelicConfigPath)
...
+            var unpackedComponents = GetUnpackedComponents(unpackedLocation);
+
+            ValidationHelpers.ValidateComponents(expectedComponents, unpackedComponents, "Agent Nuget");
+
+            FileHelpers.DeleteDirectories(unpackedLocation);
+        }
+
+        private string Unpack()
+        {
+            var unpackDir = $@"{OutputDirectory}\unpacked";
```
Comment (nr-ahemsath) [inline build/ArtifactBuilder/Artifacts/NugetAgent.cs:123]:
```suggestion
            var unpackDir = Path.Join(OutputDirectory, "unpacked");
```

File: build/ArtifactBuilder/Artifacts/NugetAgent.cs [hunk 2]
```diff
@@ -95,5 +99,99 @@ private void TransformNewRelicConfig(string newRelicConfigPath)
...
+            var nugetFile = $@"{OutputDirectory}\NewRelic.Agent.{_frameworkAgentComponents.Version}.nupkg";
```
Comment (nr-ahemsath) [inline build/ArtifactBuilder/Artifacts/NugetAgent.cs:124]:
```suggestion
            var nugetFile = Path.Join(OutputDirectory, $"NewRelic.Agent.{_frameworkAgentComponents.Version}.nupkg");
```

File: build/ArtifactBuilder/Artifacts/NugetAgentApi.cs [hunk 3]
```diff
@@ -1,30 +1,106 @@
...
+            var unpackedComponents = GetUnpackedComponents(unpackedLocation);
+
+            ValidationHelpers.ValidateComponents(expectedComponents, unpackedComponents, "Agent Api Nuget");
+
+            FileHelpers.DeleteDirectories(unpackedLocation);
+        }
+
+        private string Unpack()
+        {
+            var unpackDir = $@"{OutputDirectory}\unpacked";
```
Comment (nr-ahemsath) [inline build/ArtifactBuilder/Artifacts/NugetAgentApi.cs:55]:
```suggestion
            var unpackDir = Path.Join(OutputDirectory, "unpacked");
```

File: build/ArtifactBuilder/Artifacts/NugetAgentApi.cs [hunk 4]
```diff
@@ -1,30 +1,106 @@
...
+            var nugetFile = $@"{OutputDirectory}\NewRelic.Agent.Api.{_frameworkAgentComponents.Version}.nupkg";
```
Comment (nr-ahemsath) [inline build/ArtifactBuilder/Artifacts/NugetAgentApi.cs:56]:
```suggestion
            var nugetFile = Path.Join(OutputDirectory}, $"NewRelic.Agent.Api.{_frameworkAgentComponents.Version}.nupkg");
```

**Review by nr-ahemsath — COMMENTED** [review]
I've made some suggestions on using Path.Join in a few places, otherwise looks good.

**Review by nr-ahemsath — APPROVED** [review]
👍

**Review by jaffinito — APPROVED** [review]

**Comment by nr-ahemsath** [comment]
Merging despite not having all integration tests pass; these Artifactbuilder test changes do not affect the agent code contents.

**Pull request merged**